### ISSUE TITLE
Issue with Unity Debugger using VSCode - error while processing request 'setBreakpoints' (exception: Cannot perform member binding on null' 

### ISSUE THREAD

**Issue by weinand** [Unity-Technologies/vscode-unity-debug#13]
_From @NalinNexus on July 13, 2016 11:35_

- VSCode Version:1.31
- OS Version: Mac OSX Latest

Steps to Reproduce:

1.  try debugging project in Unity VSCode
2. set breakpoints, press play in VSCode
3. error below comes up, meaning no breakpoints can ever be hit during debug session

error while processing request 'setBreakpoints' (exception: Cannot perform member binding on null' value) Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: Cannot perform member binding onnull' value
at (wrapper dynamic-method) System.Object:CallSite.Target (System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,object)
at System.Dynamic.UpdateDelegates.UpdateAndExecute1T0,TRet <0x1145f04d0 + 0x005b5> in :0 
at OpenDebug.Utilities.ExpandVariables (System.String format, System.Object variables, Boolean underscoredOnly) <0x1146c3a90 + 0x001e6> in :0 
at (wrapper dynamic-method) System.Object:CallSite.Target (System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Type,string,object)
at System.Dynamic.UpdateDelegates.UpdateAndExecute3T0,T1,T2,TRet <0x1145ea730 + 0x00643> in :0 
at OpenDebug.Responder.SetBody (System.Object body) <0x1146c02e0 + 0x0033a> in :0 
at UnityDebug.Program+c__AnonStorey0.<>m__0 (System.String command, System.Object args, IResponder responder) <0x1145d9a40 + 0x00ac9> in :0 UnityDebug: Attached to Unity process 'Unity Editor (Unity)' (2614)

_Copied from original issue: Microsoft/vscode#9204_

**Comment by yanjingzhaisun** [comment]
Try download unity debugger in VSCode extensions menu instead of using the link provided by Unity or UnityVSCode plugin. 
<img width="358" alt="screen shot 2016-07-14 at 6 57 58 pm" src="https://cloud.githubusercontent.com/assets/6327230/16858454/ef98651e-49f4-11e6-82f5-4c80a50c465f.png">

**Comment by NalinNexus** [comment]
@yanjingzhaisun thank you so much for this

it fixed the problem 👍 :)

**Comment by nrj** [comment]
@yanjingzhaisun this problem is still occurring for me. I tried uninstalling the "Debugger for Unity" and reinstalling via the VSCode plugin but I still cannot set any breakpoints.

```
error while processing request 'setBreakpoints' (exception: Cannot perform member binding on `null' value)
Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: Cannot perform member binding on `null' value
  at (wrapper dynamic-method) System.Object:CallSite.Target (System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,object)
  at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet] (System.Runtime.CompilerServices.CallSite site, System.Dynamic.T0 arg0) <0x3027000 + 0x00406> in <filename unknown>:0 
  at OpenDebug.Utilities.ExpandVariables (System.String format, System.Object variables, Boolean underscoredOnly) <0x3380020 + 0x0016b> in <filename unknown>:0 
  at (wrapper dynamic-method) System.Object:CallSite.Target (System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Type,string,object)
  at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet] (System.Runtime.CompilerServices.CallSite site, System.Dynamic.T0 arg0, System.Dynamic.T1 arg1, System.Dynamic.T2 arg2) <0x3023400 + 0x00474> in <filename unknown>:0 
  at OpenDebug.Responder.SetBody (System.Object body) <0x337dba0 + 0x00287> in <filename unknown>:0 
  at UnityDebug.Program+<Dispatch>c__AnonStorey0.<>m__0 (System.String command, System.Object args, IResponder responder) <0x3017000 + 0x008b5> in <filename unknown>:0 UnityDebug: Attached to Unity process 'Unity Editor (Unity)' (4126)
```

![screen shot 2016-07-26 at 11 47 33](https://cloud.githubusercontent.com/assets/149472/17133532/c901420a-5326-11e6-9bd7-5059d34cbb44.png)

**Comment by lukaszunity** [comment]
@nrj 

I'm unable to reproduce this issue. You should be able to update the Debugger for Unity to version 1.0.3. I see version 1.0.1 in your screenshot.

You can try manually deleting the extension and then re-installing it.

Install Paths:
OSX/Linux: ${HOME}/.vscode/extensions/Unity.unity-debug-x.y.z
Windows: %USERPROFILE%\.vscode\extensions\Unity.unity-debug-x.y.z

**Comment by thurn** [comment]
Had the same issue but installing directly from within VSCode solved the problem. Maybe just remove the link from Unity if it doesn't work?

**Comment by lukaszunity** [comment]
@thurn Which link are you referring to?

**Comment by thurn** [comment]
The "install unity debugger" button in the vscode settings: https://imgur.com/a/hwhTT

**Comment by lukaszunity** [comment]
Oh, I see. That button is from the https://github.com/dotBunny/VSCode editor plugin. @reapazor, would it be possible fix this issue with Unity Debugger extension not working when install from your editor plugin or would possibly inform users that they should install the extension from VS Code instead?

**Comment by reapazor** [comment]
I'll push an update to GIT today, though the UAS update ... Well it can take a while for approval ;)

--
Matthew Davey

On Aug 22, 2016, at 02:34, Lukasz <notifications@github.com<mailto:notifications@github.com>> wrote:


Oh, I see. That button is from the https://github.com/dotBunny/VSCode editor plugin. @reapazor<https://github.com/reapazor>, would it be possible fix this issue with Unity Debugger extension not working when install from your editor plugin or would possibly inform users that they should install the extension from VS Code instead?

-

**Comment by lukaszunity** [comment]
Great, thanks ;)

**Comment by reapazor** [comment]
Looks like they added a way for me to install the plugin from command line even :) so thats gonna happen.
---
Matthew Davey
dotBunny Inc.

888.632.1993 x700

On Aug 22, 2016, at 7:45 AM, Lukasz <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #13<https://github.com/Unity-Technologies/vscode-unity-debug/issues/13>.

**Comment by reapazor** [comment]
I can’t for the life of me get it to use its own command line arguments ;/ *sigh* ill just update the package in the cloud :D
---
Matthew Davey
dotBunny Inc.

888.632.1993 x700

On Aug 22, 2016, at 11:18 AM, Matthew Davey <matthew.davey@dotbunny.com<mailto:matthew.davey@dotbunny.com>> wrote:

Looks like they added a way for me to install the plugin from command line even :) so thats gonna happen.
---
Matthew Davey
dotBunny Inc.

888.632.1993 x700

On Aug 22, 2016, at 7:45 AM, Lukasz <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #13<https://github.com/Unity-Technologies/vscode-unity-debug/issues/13>.

**Comment by SilhouettesForYou** [comment]
### [Attach to Process Picker](https://marketplace.visualstudio.com/items?itemName=Unity.unity-debug)
* In the command palette type "Unity Attach Debugger"
* Wait a bit for the Unity processes list to appear at the top of the VS Code window.
* Select the Unity process you wish to attach the debugger to.
